### PR TITLE
[4.0] RavenDB-11908 - We want to report CPU percentage that are based on the number of cores we actually use

### PR DESCRIPTION
**Pull request by grisha-kotler**
http://issues.hibernatingrhinos.com/issue/RavenDB-11908

### CODE REVIEW

File: src/Raven.Server/Utils/CpuUsage.cs [hunk 1]
```diff
@@ -329,62 +267,154 @@ private static unsafe MacInfo GetMacInfo()
...
+                return Bits.NumberOfSetBits(process.ProcessorAffinity.ToInt64());
+            }
+            catch (NotSupportedException)
+            {
+                return ProcessorInfo.ProcessorCount;
+            }
+            catch (Exception e)
+            {
+                if (Logger.IsInfoEnabled)
+                    Logger.Info($"Failure to get the number of active cores, error: {e.Message}", e);
```
Comment (ayende) [inline src/Raven.Server/Utils/CpuUsage.cs]: There is not need to include the `Message` here, it is already in the exception

File: src/Raven.Server/Utils/CpuUsage.cs [hunk 2]
```diff
@@ -46,24 +48,43 @@ public static (double MachineCpuUsage, double ProcessCpuUsage) Calculate()
...
+        {
+            var processorTimeDiff = currentInfo.TotalProcessorTimeTicks - previousInfo.TotalProcessorTimeTicks;
+            var timeDiff = currentInfo.TimeTicks - previousInfo.TimeTicks;
+            if (timeDiff <= 0)
+            {
+                //overflow
+                return _lastCpuInfo?.ProcessCpuUsage ?? 0;
+            }
+
+            var processCpuUsage = (processorTimeDiff * 100.0) / timeDiff / currentInfo.ActiveCores;
```
Comment (ayende) [inline src/Raven.Server/Utils/CpuUsage.cs:89]: Need to check that `ActiveCores` isn't zero either

Comment (grisha-kotler) [reply src/Raven.Server/Utils/CpuUsage.cs:89]: done

**Comment by ppekrol** [comment]
@grisha-kotler can you resolve conflicts?

**Comment by grisha-kotler** [comment]
done

**Pull request merged**